pkg/controllers/provisioning/v1alpha1/reallocator/collection.go

```diff
@@ -0,0 +1,66 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reallocator
+
+import (
+	"context"
+	"fmt"
+	"github.com/awslabs/karpenter/pkg/controllers/provisioning/v1alpha1"
+	"go.uber.org/zap"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+)
+
+type Collector struct {
```

Comment:
I think you're using the word collector w.r.t garbage collection. I might instead use the term terminator since garbage collection means something else in k8s-land

Reply:
Ah. I had terminator, but then thought collector might be clever with that exact reason. I'll revert to terminator.